app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -144,10 +145,51 @@ public void afterTextChanged(Editable s) {
         }
     }
 
+    /**
+     * Uses previously saved theme if there is any, if not then uses default.
+     * Adds preference theme listener and saves value chosen by user to shared preferences
+     * to remember later
+     */
+    private void prepareTheme() {
```

Comment:
now you can delete all of this except 
```
setOnPreferenceChangeListener{
activity.recreate()
}
```

Reply:
I removed the irrelevant code now but the if - else clause is important to set the initial summary so I kept it.